packages/yourloops/pages/profile/profile.tsx

```diff
@@ -161,50 +136,32 @@ const ProfilePage = (props: ProfilePageProps): JSX.Element => {
 
   const role = user.role;
   const showFeedback = role === UserRoles.hcp;
-
-  const [firstName, setFirstName] = React.useState<string>(getUserFirstName(user));
-  const [lastName, setLastName] = React.useState<string>(getUserLastName(user));
-  const [currentPassword, setCurrentPassword] = React.useState<string>("");
-  const [password, setPassword] = React.useState<string>("");
-  const [passwordConfirmation, setPasswordConfirmation] = React.useState<string>("");
-  const [unit, setUnit] = React.useState<Units>(user.settings?.units?.bg ?? Units.gram);
-  const [birthDate, setBirthDate] = React.useState<string>(user.profile?.patient?.birthday ?? "");
-  const [switchRoleOpen, setSwitchRoleOpen] = React.useState<boolean>(false);
-  const [lang, setLang] = React.useState<LanguageCodes>(user.preferences?.displayLanguageCode ?? getCurrentLang());
-  const [hcpProfession, setHcpProfession] = React.useState<HcpProfession>(user.profile?.hcpProfession ?? HcpProfession.empty);
-  const [feedbackAccepted, setFeedbackAccepted] = React.useState(Boolean(user?.profile?.contactConsent?.isAccepted));
-
-  let passwordCheckResults: CheckPasswordStrengthResults;
-  if (password.length > 0) {
-    passwordCheckResults = checkPasswordStrength(password);
-  } else {
-    passwordCheckResults = { onError: false, helperText: "", score: -1 };
-  }
-
-  React.useEffect(() => {
-    // To be sure we have the locale:
-    if (!availableLanguageCodes.includes(lang)) {
-      setLang(getCurrentLang());
-    }
-    setPageTitle(t("account-preferences"));
-  }, [lang, t]);
-
-  React.useEffect(() => {
-    // ISO date format is required from the user: It's not a very user-friendly format in all countries, We should change it
-    if (role === UserRoles.patient && !!birthDate) {
-      let birthday = birthDate;
-      if (birthday.length > 0 && birthday.indexOf("T") > 0) {
-        birthday = birthday.split("T")[0];
-      }
-      if (REGEX_BIRTHDATE.test(birthday)) {
-        setBirthDate(birthday);
-      } else {
-        setBirthDate("");
-      }
-    }
-    // No deps here, because we want the effect only when the component is mounting
-    // If we set the deps, the patient won't be able to change its birthday.
-  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+  const [firstName, setFirstName] = useState<string>(user.firstName);
```

Comment:
Suggestion: Having some many state does not ease code understandabilty and probably causes some excessive re-redeering. I propose slicing the states in objects (one object for each sub component). 
Contact me on slack if my comment is not clear!

Reply:
For this case, even if each component has its own state, it's the Profile component that merge all states and call the hook, the problem will be the same and child components wild be re-rendered. But it's a good point for futur components ;)